src/Native/build-native.sh

```diff
@@ -87,19 +88,29 @@ check_native_prereqs()
     # Check presence of CMake on the path
     hash cmake 2>/dev/null || { echo >&2 "Please install cmake before running this script"; exit 1; }
 
+    if [ "$__GccBuild" = 0 ]; then
+        # Minimum required version of clang is version 3.9 for arm/armel cross build
+        if [ "$__CrossBuild" = 1 ] && { [ "$__BuildArch" = "arm" ] || [ "$__BuildArch" = "armel" ]; }; then
+            if [ "$__ClangMajorVersion" -lt 3 ] || { [ "$__ClangMajorVersion" -eq 3 ] && [ "$__ClangMinorVersion" -lt 9 ]; }; then
+                echo "Please install clang3.9 or latest for arm/armel cross build"; exit 1;
+            fi
+        fi
 
-    # Minimum required version of clang is version 3.9 for arm/armel cross build
-    if [[ $__CrossBuild == 1 && ("$__BuildArch" == "arm" || "$__BuildArch" == "armel") ]]; then
-        if ! [[ "$__ClangMajorVersion" -gt "3" || ( $__ClangMajorVersion == 3 && $__ClangMinorVersion == 9 ) ]]; then
-            echo "Please install clang3.9 or latest for arm/armel cross build"; exit 1;
+        # Check for clang
+        hash "clang-$__ClangMajorVersion.$__ClangMinorVersion" 2>/dev/null || hash "clang$__ClangMajorVersion$__ClangMinorVersion" 2>/dev/null ||  hash clang 2>/dev/null || { echo >&2 "Please install clang before running this script"; exit 1; }
+    else
+        # Minimum required version of gcc is version 5.0 for arm/armel cross build
+        if [ "$__CrossBuild" = 1 ] && { [ "$__BuildArch" = "arm" ] || [ "$__BuildArch" = "armel" ]; }; then
+            if [ "$__GccMajorVersion" -lt 5 ]; then
+                echo "Please install gcc version 5 or latest for arm/armel cross build"; exit 1;
```

Comment:
gcc v5 (on Debian 8 from https://hub.docker.com/_/gcc) is the lowest version where it worked and v8 (on FreeBSD) is the highest.

I have tested with gcc 4.4, 4.6, 4.7, 4.9, on (old) CentOS 5.11; they all failed due to few missing compiler options and C11's `_Generic` usage.